dir.common.props

```diff
@@ -56,6 +56,14 @@
     <BinDir>$(__BinDir)\</BinDir>
     <BinDir Condition="'$(__BinDir)'==''">$(RootBinDir)Product\$(PlatformConfigPathPart)\</BinDir>
 
+    <!-- Use the envvar __IntermediatesDir to find the location of the obj   -->
+    <!-- dir                                                                 -->
+    <!-- Note that if nmake is used, the obj/ folder is nmakeobj/ therefore  -->
+    <!-- env variable is the only source of truth. This is only done for the -->
+    <!-- windows formatting jobs which use nmake and clang-format to build   -->
+    <!-- the product.                                                        -->
+    <__IntermediatesDir Condition="'$(__IntermediatesDir)' == ''">$(RootBinDir)obj/$(BuildOS).$(BuildArch).$(BuildType)/</__IntermediatesDir>
```

Comment:
Instead of relying on the environment variable, can we just have some property (e.g. `IbcMergeVersionOutputFile`, `DumpPackageVersionOutputFile`, or something similar) that is used by the `Dump*PackageVersion` targets and should be explicitly set when running them?

I'd much prefer the clear contract of the caller setting exactly where the file should be written and then reading from it instead of the calling script expecting/knowing that the project uses a specific environment variable and file name.

Reply:
Addressed